tools/ossf_best_practices/silver_criteria.md

```diff
@@ -0,0 +1,338 @@
+
+**N/A**
+
+> The software produced by the project SHOULD support secure protocols for all of its network communications, such as SSHv2 or later, TLS1.2 or later (HTTPS), IPsec, SFTP, and SNMPv3. Insecure protocols such as FTP, HTTP, telnet, SSLv3 or earlier, and SSHv1 SHOULD be disabled by default, and only enabled if the user specifically configures it. If the software produced by the project does not support network communications, select "not applicable" (N/A). 
+
+_Possible answers: Met/Unmet/NA_
+
+> The software produced by the project SHOULD, if it supports or uses TLS, support at least TLS version 1.2. Note that the predecessor of TLS was called SSL. If the software does not use TLS, select "not applicable" (N/A).
+
+**Met**
+
+> The software produced by the project MUST, if it supports TLS, perform TLS certificate verification by default when using TLS, including on subresources. If the software does not use TLS, select "not applicable" (N/A).
+
+_Possible answers: Met/Unmet/NA_
+
+> The software produced by the project MUST, if it supports TLS, perform certificate verification before sending HTTP headers with private information (such as secure cookies). If the software does not use TLS, select "not applicable" (N/A).
+
+_Possible answers: Met/Unmet/NA_
+
+## Secure release
+
+> The project MUST cryptographically sign releases of the project results intended for widespread use, and there MUST be a documented process explaining to users how they can obtain the public signing keys and verify the signature(s). The private key for these signature(s) MUST NOT be on site(s) used to directly distribute the software to the public. If releases are not intended for widespread use, select "not applicable" (N/A).
+
+**Met** https://github.com/nodejs/node/blob/main/doc/contributing/releases.md#3-a-publicly-listed-gpg-key
+
+> It is SUGGESTED that in the version control system, each important version tag (a tag that is part of a major release, minor release, or fixes publicly noted vulnerabilities) be cryptographically signed and verifiable as described in [signed_releases](https://bestpractices.coreinfrastructure.org/en/projects/29?criteria_level=1#signed_releases).
+
+**Met**
+https://github.com/nodejs/node/blob/main/doc/contributing/releases.md#11-tag-and-sign-the-release-commit
+
+## Other security issues
+
+> The project results MUST check all inputs from potentially untrusted sources to ensure they are valid (an *allowlist*), and reject invalid inputs, if there are any restrictions on the data at all.
+
+**Met**
+
+> Hardening mechanisms SHOULD be used in the software produced by the project so that software defects are less likely to result in security vulnerabilities.
+
+_Possible answers: Met/Unmet/NA_
```

Comment:
Not sure if this apply to Node 🤔

Reply:
Shouldn't it be the isolated build machines?

Reply:
Is there a definition of `Hardening mechanisms`, I don't think the use of isolated build machines is `in the software`

Reply:
I assume this refers to hardening mechanisms such as `-fsanitize=bounds`, `-fstack-protector`, etc. Some are enabled by default but most come with significant performance overheads, thus disabled.

Reply:
So, based on [this](https://github.com/coreinfrastructure/best-practices-badge/blob/f4d207721526c1f60c2003f747a46ae897ad9977/doc/other.md?plain=1#L1070):

> Hardening mechanisms may include HTTP headers like Content Security Policy (CSP), compiler flags to mitigate attacks (such as -fstack-protector), or compiler flags to eliminate undefined behavior. For our purposes least privilege is not considered a hardening mechanism (least privilege is important, but separate).

I think this is a `N/A` in our case, because the final user has the power to decide. As well we clarified this in [our thread model](https://github.com/nodejs/node/blob/main/SECURITY.md#the-nodejs-threat-model)

Reply:
> I think this is a N/A in our case, because the final user has the power to decide.

@UlisesGascon Not all of the mentioned examples are up to the user, e.g., "compiler flags to mitigate attacks (such as -fstack-protector), or compiler flags to eliminate undefined behavior" (which is what I referred to in https://github.com/nodejs/security-wg/pull/955#discussion_r1255662698).

Reply:
Maybe N/A and mention those cases?

Reply:
> Maybe N/A and mention those cases?

Agree, do we have a list of the relevant flags so we can include a URL or similar?

Reply:
I don't know any specific list. The exact compiler flags likely vary depending on the target operating system, the compiler, the build type (debug/release/asan/...), etc.

Reply:
What about:

> N/A - However, specific compiler flags such as `-fstack-protector` might not be passed to the distributed binary

Also, might be worth it to check other runtimes answer (if they have this initiative)

Reply:
Quick update: In the last team meeting #1059 we agreed to use "N/A". We can review it again in the next iteration if needed :)